Add documentation for exposing Kiali and Grafana with oauth2_proxy (#11401)
kyma-project/kyma: docs/operation-guides/config-07-kiali-grafana.md

            timeoutSeconds: 2
          readinessProbe:
            httpGet:
              path: /oauth2/healthy
              port: http
            initialDelaySeconds: 3
            timeoutSeconds: 2
        securityContext:
          fsGroup: 65534
          runAsNonRoot: true
          runAsUser: 65534
  ```

  5. Create a Service for the `oauth2_proxy`:

  ```yaml
  apiVersion: v1
  kind: Service
  metadata:
    name: oauth2-kiali
    labels:
      app: oauth2-kiali
  spec:
    type: ClusterIP
    ports:
    - port: 3000
      name: http
      protocol: TCP
      targetPort: http
    selector:
      app: oauth2-kiali
  ```

  6. Expose the Service via Istio by creating a VirtualService. Set the domain in the `hosts` list to your desired name:

  ```yaml
  apiVersion: networking.istio.io/v1alpha3
  kind: VirtualService
  metadata:
    name: oauth2-kiali
  spec:
    hosts:
    - kiali.kyma.example.com
    gateways:
    - kyma-system/kyma-gateway
    http:
    - match:
      - uri:
          regex: /.*
      route:
      - destination:
          port:
            number: 3000
          host: oauth2-kiali
  ```
  </details>
</div>
  <details>
  <summary>
  Grafana
  </summary>

>   1. Chose a domain for the exposed service under the Kyma base domain. E.g., `grafana.kyma.example.com` if your Kyma cluster is reachable under `kyma.example.com`.

Reviewer: ```suggestion
  1. Chose a domain for the exposed service under the Kyma base domain. For example, if your Kyma cluster is reachable under `kyma.example.com`, use `grafana.kyma.example.com` .
```

>   2. Create a new Github application under https://github.com/settings/apps. Set the callback URL to `https://grafana.kyma.example.com/oauth2/callback`. Ensure at least read-only access to email addresses for the Github application. Copy the client id and secret.

Reviewer: ```suggestion
  2. Create a new Github application under https://github.com/settings/apps. Set the callback URL to `https://grafana.kyma.example.com/oauth2/callback`. Ensure at least read-only access to email addresses for the Github application. Copy the client ID and secret.
```

>   3. Create a Kubernetes Secret for the client id and secret:

Reviewer: ```suggestion
  3. Create a Kubernetes Secret for the client ID and secret:
```

            initialDelaySeconds: 3
            timeoutSeconds: 2
          readinessProbe:
            httpGet:
              path: /oauth2/healthy
              port: http
            initialDelaySeconds: 3
            timeoutSeconds: 2
        securityContext:
          fsGroup: 65534
          runAsNonRoot: true
          runAsUser: 65534
  ```

  5. Create a Service for the `oauth2_proxy`:

  ```yaml
  apiVersion: v1
  kind: Service
  metadata:
    name: oauth2-grafana
    labels:
      app: oauth2-grafana
  spec:
    type: ClusterIP
    ports:
    - port: 3000
      name: http
      protocol: TCP
      targetPort: http
    selector:
      app: oauth2-grafana
  ```

  6. Expose the Service via Istio by creating a VirtualService. Set the domain in the `hosts` list to your desired name:

  ```yaml
  apiVersion: networking.istio.io/v1alpha3
  kind: VirtualService
  metadata:
    name: oauth2-grafana
  spec:
    hosts:
    - grafana.kyma.example.com
    gateways:
    - kyma-system/kyma-gateway
    http:
    - match:
      - uri:
          regex: /.*
      route:
      - destination:
          port:
            number: 3000
          host: oauth2-grafana
  ```
  </details>
</div>

> The `oauth2_proxy` supports a wide range of other well known authentication services or OpenID Connect for custom solutions. See the [list of supported providers](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider) to find instructions for other authentication services.

Reviewer: ```suggestion
>**NOTE:** The `oauth2_proxy` supports a wide range of other well-known authentication services or OpenID Connect for custom solutions. See the [list of supported providers](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider) to find instructions for other authentication services.
```

I wonder whether this information would be better above the step description, as it explains why we call our instructions an "example".  Wouldn't readers want to know this first; before reading all the steps?

Author: Good suggestion. Will move it.
